packages/domain/src/entities/Publication.ts

```diff
@@ -29,6 +29,8 @@ export enum Erc20ComparisonOperator {
 /**
  * The active wallet must satisfy the NFT ownership criteria
  */
+// TODO: Introduce a new type that splits NFT ownership criteria into two types: one for ERC721 and one for ERC1155
+// This would allow to force the `tokenIds` field to be always presented for ERC1155
```

Comment:
Why not removing the `?` from `tokenIds`?

Reply:
The ERC721 can be called without any sort of `id` to just check if the wallet has any nft from the collection.
I tried to split it  by `type NFTOwnershipCriterion = ERC721OwnershipCriterion | ERC1155OwnershipCriterion` but was causing an issues with `NftOwnershipCriterionSchema` which would require it to be also an union.
But due to the `type` being inside inside `z.discriminatedUnion` i can't repeat the same type twice in the schema.
One approach to consider at some point is to split `DecryptionCriteriaType.NFT_OWNERSHIP` by `DecryptionCriteriaType.ERC721_OWNERSHIP` and `DecryptionCriteriaType.ERC1155Ownership` which i think is more clear from the consumer side as you can at least see that quite likely different contract methods will be called to check the ownership.

Reply:
Interesting, will look into token gated support for metadata v3 soon. Will include your suggestion.